### PR TITLE
added

### PR DESCRIPTION
**Pull request by OlegKandalov**
repo:
- Discount, VendorLocation, Tag, UserLocation, Vendor
service:
- UserLocation
converter:
- UserLocationConverter

### CODE REVIEW

File: src/main/java/com/exadel/discount/platform/converter/UserLocationConverter.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package com.exadel.discount.platform.converter;
+
+import com.exadel.discount.platform.model.UserLocation;
+import com.exadel.discount.platform.model.dto.UserLocationDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class UserLocationConverter {
+
+    public static UserLocationDto entityToDto(UserLocation userLocation) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/converter/UserLocationConverter.java:10]: There's ModelMapper for this http://modelmapper.org/getting-started/

Comment (ohalushchak-exadel) [reply src/main/java/com/exadel/discount/platform/converter/UserLocationConverter.java:10]: and turn it into a `@Component`

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    public List<UserLocationDto> getAll() {
+        List<UserLocationDto> users = new ArrayList<>();
+        for (UserLocation u : userLocationRepository.findAll()) {
+            users.add(UserLocationConverter.entityToDto(u));
+        }
+        return users;
+    }
+
+    @Override
+    public UserLocationDto add(UserLocationDto userLocationDto) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:34]: save() or create()

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    @Autowired
+    public UserLocationServiceImpl(UserLocationRepository userLocationRepository) {
+        this.userLocationRepository = userLocationRepository;
+    }
+
+    @Override
+    public List<UserLocationDto> getAll() {
+        List<UserLocationDto> users = new ArrayList<>();
+        for (UserLocation u : userLocationRepository.findAll()) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:27]: create a method in mapper to convert a collection

Comment (ohalushchak-exadel) [reply src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:27]: also using streams would be more convenient here

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @Override
+    public void update(UUID id, UserLocationDto userLocationDto) {
+        UserLocation userLocation = userLocationRepository.findById(id).orElseThrow();
+        //userLocation.setUserDetails(userLocationDto.getUserDetails());
+        userLocation.setCountry(userLocationDto.getCountry());
+        userLocation.setCity(userLocationDto.getCity());
+        userLocation.setAddressLine(userLocationDto.getAddressLine());
+        userLocation.setDeleted(userLocationDto.isDeleted());
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:52]: do this in a mapper

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+        userLocationRepository.save(userLocation);
+    }
+
+    @Override
+    public void toArchive(UUID id) {
+        UserLocation location = userLocationRepository.findById(id).orElseThrow();
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:59]: fine for now but you'll need to throw an appropriate exception which you can handle in a ControllerAdvice and return a 404 error to the client

File: src/main/resources/liquibase/0001_initial_schema.xml [hunk 6]
```diff
@@ -27,6 +38,9 @@
                 <constraints nullable="false" unique="true"/>
             </column>
             <column name="u_password" type="VARCHAR(170)"/>
+            <column name="u_user_role" type="USER_ROLE">
```
Comment (ohalushchak-exadel) [inline src/main/resources/liquibase/0001_initial_schema.xml:41]: Changes to the existing changesets will break the deployment unless the database is completely clean. Such changes are a a big NO-NO in the applications which reached production. 
For now just wipe the existing database on heroku so you can deploy this version

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java (hunk 2 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:34]: done

File: src/main/java/com/exadel/discount/platform/converter/UserLocationConverter.java (hunk 1 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/converter/UserLocationConverter.java:10]: done

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java (hunk 3 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:27]: done

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java (hunk 4 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:52]: done

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java (hunk 5 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:59]: okey thank you

File: src/main/java/com/exadel/discount/platform/converter/UserLocationConverter.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import java.lang.reflect.Type;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class UserLocationConverter extends ModelMapper {
+    private final ModelMapper modelMapper;
+
+    public UserLocationConverter(ModelMapper modelMapper) {
+        this.modelMapper = modelMapper;
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/converter/UserLocationConverter.java:17]: create Mapper in the constructor

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class UserLocationServiceImpl implements UserLocationService {
+    private final UserLocationRepository userLocationRepository;
+    private final UserLocationConverter converter;
+
+    @Autowired
+    public UserLocationServiceImpl(UserLocationRepository userLocationRepository, UserLocationConverter converter) {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:21]: just use `@RequiredArgsConstructor` on class

File: src/main/java/com/exadel/discount/platform/service/interfaces/Service.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.exadel.discount.platform.service.interfaces;
+
+import java.util.List;
+import java.util.UUID;
+
+public interface Service<T> {
```
Comment (ohalushchak-exadel) [inline src/main/java/com/exadel/discount/platform/service/interfaces/Service.java:6]: Name is way too generic. Name it CrudService at least

File: src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java (hunk 8 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/service/UserLocationServiceImpl.java:21]: done

File: src/main/java/com/exadel/discount/platform/service/interfaces/Service.java (hunk 9 above)
Comment (OlegKandalov) [reply src/main/java/com/exadel/discount/platform/service/interfaces/Service.java:6]: done

**Pull request closed without merging**